Update cleanup directory defaults

gwleuverink opened the pull request:
Accompanies PR https://github.com/NativePHP/electron/pull/153

`storage/app/framework/{sessions,testing,cache}` & `storage/logs/laravel.log` have been moved to the build command default exclude patterns. Since They should aways be excluded from the final build because of security concerns.

I've added `vendor/bin` & `node_modules` as sensible defaults to be tweaked.

@simonhamp What do you think about this? If accepted this should be mentioned in a update guide.

simonhamp reviewed (approved):
Yep this seems fine. I'm ignoring the failing test as it won't apply in the context where this new config gets used, but I can't merge and release this until that Minify command is gone

gwleuverink commented:
I see. I can delete the minify command aswel. The `IgnoreFilesAndFoldersTest` should at least be ported over to the other PR i think. Will pick that up!

simonhamp commented:
> This does raise the question of where things belong.

Yeh I've found this to be a bit of a challenge. I think this stems from the fact that only have this Laravel adapter and the Electron driver. It may become clearer if/when we add more drivers and adapters...

I have thoughts on how we may divide this better, but I'm keen not to over-engineer stuff on this front until it becomes more obvious.

Maybe for now we do what works and refactor once it becomes clear?

simonhamp commented:
On not merging... actually, maybe we can... I don't think that minify command is even being used any more, right?

gwleuverink commented:
I just noticed something. Slight oversight on my part.

Since the app cleaning is done before running `composer install --no-dev` inside the build directory, any executables that are not dev-dependencies will be reinstalled. So even though `vendor/bin` is in the exclude list, this directory will not be empty.

Without the dev dependencies, the footprint still is very small (20kb) in a fresh app.
I'm considering removing `vendor/bin` from the config array. Or should I rewrite the thing so we do a extra pass of cleaning after composer install ran?

gwleuverink commented:
> Let's do the extra pass AND remove it from the config array

Cool! I've added that here https://github.com/NativePHP/electron/pull/153/commits/550922d31d530c125731fe172dab596a5f981d87

I've also removed `vendor/bin` from the config, but added `*/tests`. Thought that was a nice default, since most composer packages include that folder.

> Yeh I've found this to be a bit of a challenge

I moved the list of default patterns to the internal config, this way the definitions are at least inside the laravel repo. Is that okay? I can move them back.

----

I left the `content` dir in the config, but I don't think it's used at all.

SRWieZ reviewed (approved):
Already approved, just tested it again. 

Approved again!

gwleuverink commented:
Should we add this to an upgrade guide?
